gui/src/launcher.rs

```diff
@@ -177,6 +178,19 @@ async fn check_network_datadir(mut path: PathBuf, network: Network) -> Result<Ne
         )
     })?;
 
+    if let Some(exe_config) = cfg.internal_bitcoind_exe_config {
+        let args = vec![
+            format!("-chain={}", &network.to_core_arg()),
+            format!("-datadir={}", exe_config.data_dir.to_string_lossy()),
+        ];
+        info!("Starting internal bitcoind");
+        let _ = std::process::Command::new(exe_config.exe_path)
+            .args(&args)
+            .stdout(std::process::Stdio::null())
+            .spawn();
+        std::thread::sleep(std::time::Duration::from_secs(1));
```

Comment:
This is brittle. If this is to wait for `bitcoind` to start up you can just use `BitcoinD::new()` which will do just that. In addition this allows you to detect a `bitcoind` startup error in which case you can print the `stderr`?

Reply:
Without the `sleep`, I seem to get an error that the cookie file is missing. This also happens during the installer.

Reply:
Good to know, then this is racy. A more robust way of doing this would be (in pseudocode):
```rust
while !cookie_file.exists() {
     sleep(1s);
}
```

One way it could be achieved without having to compute the cookie file path yourself would be to retry on `BitcoinD` error about the cookie file. Note however that we need to limit the number of retries, to avoid an infinite loop if the bitcoind process dies for whatever reason.

Reply:
Interesting edge case that came up when testing: if the error is that `bitcoind` is already running:
> Error: Cannot obtain a lock on data directory /home/darosior/projects/wizardsardine/liana/gui/aa/bitcoind_datadir. Bitcoin Core is probably already running.

We should continue, otherwise the user would be in an infinite loop. I don't think there is a specific return code for this error, so we would unfortunately have to match on the error string.. Something like `e.to_string().contains("is probably already running.")`.